### ISSUE TITLE
Backup w/o App Backup from GrapheneOS not recognized for Automatic Restore

### ISSUE THREAD

**Issue by dan9er** [stevesoltys/backup#599]
I'm a recent emigrate from [GrapheneOS 2023100300](https://grapheneos.org/releases#2023100300) to [CalyxOS 5.1.0](https://calyxos.org/news/2023/11/09/november-security-update/) on my Pixel 7 (`panther`).

Sequence of events:
1. I used in-app data export for the apps that had it.
2. I disabled App Backup in Seedvault to make transferring the backup folder faster for the move.
3. I deleted & remade the Seedvault folder on my phone, to have a fresh backup without the APKs from my previous backup (I had already copied and saved that off-device).
4. Ran Seedvault and copied the new backup folder to a USB (and also saved that off-device).
5. Flashed CalyxOS. Point of no return.
6. During initial setup, I restored the backup from my USB. Of course, it complained about non-existant apps, but I figured I could just install them again and Automatic Restore will take care of the rest.
7. When I opened Seedvault, *it reported that I had never made a backup*. The apps I knew I had backed up *didn't appear* in the Backup Status list. I tried to install an app I knew was in the backup, and *no data was restored*.
8. I tried switching to local storage backup and then deleting and replacing the backup folder with the one I saved. *Seedvault still can't see it and thinks I never made a backup.*
9. I tried making a new backup, the existing apps were successful but the apps from the existing backup, the ones I wanted to restore, *still didn't show up*.

It's as if the backup I just restored from in initial setup didn't exist anymore. I wasn't prompted to replace my recovery code at any point.

There is some app data I have in that backup for apps that don't have a import/export option. In my case, losing that data would be a major inconvenience, but not devastating. (Good thing I thought to do the exports separately, *then* I would've been screwed and pissed!) But for others it might not!

I'm not sure which series of actions would be useful to generate & send logs for.

**Comment by grote** [comment]
The critical point is 6. This sounds like Seedvault did find your backup and did start the restore procedure. When it could not install apps, because you turned app backup off, did you try to manually install apps before continuing? Did you ever see the restore data screen that comes after install? Was it able to restore anything like settings, wifi passwords, wallpaper, etc?

**Comment by grote** [comment]
May be related to https://github.com/seedvault-app/seedvault/pull/598

**Comment by dan9er** [comment]
I apologize, It's been a while and my memory is hazy.

> When it could not install apps, because you turned app backup off, did you try to manually install apps *before continuing*?

If there was a prompt to install the missing apps right there, it wasn't clear at all. CalyxOS did have a prompt to install some of its own apps, I didn't see any of mine there.

I was under the assumption that after setup it would continue to use that backup as it's base, so I could restore those apps "later" via Automatic Restore. So it just completely forgets about those apps' backed-up data if I don't immediately install them at setup? That's an undocumented footgun.

> Did you ever see the restore data screen that comes after install?

I think I did, however I don't remember exactly what was on it.

> Was it able to restore anything like settings, wifi passwords, wallpaper, etc?

I disabled that, since I was moving to a new ROM. I didn't want GrapheneOS specific setting data messing up my fresh CalyxOS install.

EDIT: I did backup/transfer my contacts, and that worked fine.

---

At this time I've already set up most my apps again, though there is still one app I could use for testing. Is there any way to force Seedvault to recognize the old backup, or did I get footgunned by not installing the apps during initial setup?
